core/src/main/scala/org/apache/spark/ErrorInfo.scala

```diff
@@ -73,18 +73,20 @@ private[spark] object SparkThrowableHelper {
 
   def getMessage(
       errorClass: String,
+      errorSubClass: Option[String],
```

Comment:
since this method can be called in Java, using nullable `String` is better here? e.g. https://github.com/apache/spark/pull/36693/files#diff-82cc1e88d98fb17768fdc4a36035aac5e4b4b2aa797ec5c4e14860e3be7be4ccR42

Reply:
OK, testing...